comps/llms/text-generation/predictionguard/predictionguard_llm.py

```diff
@@ -1,64 +1,54 @@
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+# TODO: Add streaming
 import os
 from fastapi.responses import StreamingResponse
-from langsmith import traceable
-from comps import GeneratedDoc, LLMParamsDoc, ServiceType, opea_microservices, register_microservice
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
 from predictionguard import PredictionGuard
-import os
-import requests
-from langsmith import traceable
+from comps import GeneratedDoc, ServiceType, opea_microservices, register_microservice
+
+# Set up Prediction Guard API key
+PG_API_KEY = os.getenv("PREDICTIONGUARD_API_KEY")
+client = PredictionGuard()
+# Define the input data model
+class LLMParamsDoc(BaseModel):
```

Comment:
Nitpick, formatting. Should have 2 blank lines around top level functions, classes, etc.